STAY IN THE LIGHT: Adds terrify Nightmare spell, terrified status effect, and a reason to mind the shadows

Rhials opened the pull request:
## About The Pull Request

Adds the Terrify spell, and its associated status effect, Terrified. This new spell is given to antagonist nightmares, as a part of their brain. The spell only works in those surrounded by darkness, and will apply the Terrified status effect if successful. Upon being Terrified, victims will passively gain **Terror Buildup** if they remain in the dark. As buildup increases, so do the negative effects, including tunnel vision, panic attacks, dizziness, and more.

There are two primary methods for mitigating terror buildup. The first is moving into the light, which will reverse the passive terror buildup and eventually make it go away. The other method is by getting a hug from a friendly hand, which will reduce buildup significantly.

Getting a hug from an UNfriendly hand (a nightmare, for instance) will cause the victim to freak out and be briefly knocked down. This can be spammed on targets who are caught alone in the dark, keeping them in an unfavorable position (sideways) and adding to the victim's terror buildup considerably. Escape into the light as soon as possible, or you'll be pushed to MAXIMUM TERROR BUILDUP.

To what end? Heart failure. Past the soft terror cap (which limits how much passively generated terror you can make) exists the hard terror cap. Bypassing that threshold will cause a stress induced heart attack and knock you unconscious (embarrassing!)

https://user-images.githubusercontent.com/28870487/209616319-e21d1815-7a59-430e-876c-4a6f62cfc766.mp4

In summary -- Nightmares can terrify victims they catch in the dark. This causes panic and makes them easy to trip up during combat. If you panic too hard, you'll have a heart attack. STAY IN THE LIGHT.
## Why It's Good For The Game

Gives Nightmares a sharp advantage when fighting single targets within their territory. Applies pressure on combatants to back out of a fight, and allows Nightmares to bully unprepared targets into submission. (I also wanted to make an objective for giving X crewmembers fear-induced heart attacks, but I don't want to be the guy who gives objectives to Nightmares).

Encourages those hunting nightmares to group up and take care of each other. Gives something interesting to play strategically around, and will hopefully present a dilemma to even the most seasoned of hunters.

Most importantly, it gives something fun for nightmares to do when they're not murdering people. Go forth and be spooky! Give people heart attacks! Be a MENACE!
## Changelog
:cl: Rhials
add: Adds the Terrify spell and Terrified status effect. Terrified victims will suffer increasingly detrimental effects until they're calmed down.
balance: Nightmares can now cast the Terrify spell on victims to plague their prey with. Attacking a Terrified human with your open hand will trip them up.
imageadd: Adds status effect and spell icons for Terrify/Terrified.
spellcheck: fixes a typo in the message for consuming a nightmare heart.
/:cl:

Rhials committed:
adds the spell

big pr wow

Rhials committed:
terrified status effect, groundwork, defines

Rhials committed:
effect sprite

better heart atack stuff, bettter examine text, new icon epic

Rhials committed:
better numbers, hug stuff

Rhials committed:
better darkness check logic, less balloon alerts

better comments better numbers better working hug interactions

better define names, better feedback and finalized terror effects.

honestly these PRs are getting done in such huge batches idk what's inspired me to go so hard on this

Rhials committed:
better text better pronouns

kills the balloon alerts in the spell because they dont like working in the darkness(?)

Rhials committed:
cool sprite

Rhials committed:
makes define names WAY better, fixes comments

Rhials committed:
eye blur fixes

Rhials committed:
jitters, better light logic

Rhials committed:
more effect reordering, better spell messages

Rhials committed:
FINALLY makes the darkness checks not suck

Rhials committed:
DO AS YOU ARE TOLD

Rhials committed:
adjustments before the rebase

Rhials committed:
Merge remote-tracking branch 'upstream/master' into terrorize

Rhials committed:
better dizzy better eye blur better FOV args

Rhials committed:
adds stacking,

Rhials committed:
undef

Rhials committed:
math adjustment

Rhials committed:
bravery check

Rhials committed:
blue hair and pronoun fix

Rhials committed:
hug logic

Rhials committed:
last touch ups

Rhials committed:
OOPS

san7890 commented on code/datums/status_effects/debuffs/terrified.dm:
```diff
@@ -0,0 +1,152 @@
+#define DARKNESS_TERROR_AMOUNT 10 //Amount of terror passively generated (or removed) on every tick based on lighting.
+#define PANIC_ATTACK_TERROR_AMOUNT 35 //How much terror a random panic attack will give the victim.
+#define HUG_TERROR_AMOUNT 60 //Amount of terror actively removed (or generated) upon being hugged.
+#define STACK_TERROR_AMOUNT 135 //Amount of terror caused by subsequent casting of the Terrify spell.
+
+#define DARKNESS_TERROR_CAP 400 //The soft cap on how much passively generated terror you can have. Takes about 30 seconds to reach without the victim being actively terrorized.
+
+#define TERROR_FEAR_THRESHOLD 140 //The terror_buildup threshold for minor fear effects to occur.
+#define TERROR_PANIC_THRESHOLD 300 //The terror_buildup threshold for the more serious effects. Takes about 20 seconds of darkness buildup to reach.
+#define TERROR_HEART_ATTACK_THRESHOLD 600 //If you actively terrorize someone already at the darkness threshold, you can cause a heart attack and knock them out.
```
transition all of these to dmdoc :)

```txt
/// Amount of terror passively generated (or removed) on every tick based on lighting.
#define DARKNESS_TERROR_AMOUNT 10
```

san7890 commented on code/modules/spells/spell_types/pointed/terrorize.dm:
```diff
@@ -0,0 +1,40 @@
+	. = ..()
+
+	if(!ishuman(cast_on))
+		to_chat(owner, span_notice("[cast_on] cannot be terrorized!"))
```
might be nice to have this be a balloon alert like "can't be terrorized!"? seems more useful to me

Rhials committed:
dee em dock

Rhials committed:
im not sure if those spaces were needed but im not waiting until the next batch of reviews to find out

san7890 commented:
061a11506b5c41a13107512c22e11e96f099d038 - i don't believe the spaces are needed but i do it that way because it looks cleaner to my eyes. whatever really

ATH1909 commented:
what's the duration on this? does it go away when your terror hits 0?

private-tristan commented:
terrify -> hug -> shove is a 2 second stun 👀

IndieanaJones commented:
While Nightmare does need buffs to better enable him to do his job, this probably won't do much of anything. Nobody hunting nightmares will be travelling without a light, dragging a light, or using miner lights, so the chance to actually get to use this only really applies to people the nightmare can easily kill as it is now.

That being said, this is flavorful and fun. I like it.

KittyNoodle commented:
make people cry when they have panic attacks

Shadowflame909 commented:
The true nightmare counter is not powergaming.

It's bar rp

Bluedino1025 commented:
It’d be neat if phobia’s also caused terror build up

Rhials committed:
fix fix fix

Rhials committed:
Update terrified.dm

Rhials committed:
oh that doesnt work oops

Rhials committed:
balloonma

Rhials committed:
unrelated typo fix

Serxule commented:
i think the rest of this find, but having hugs basically be a telebaton doesnt sound very balanced

Fikou committed:
Merge branch 'master' into terrorize

IndieanaJones commented:
> i think the rest of this find, but having hugs basically be a telebaton doesnt sound very balanced

The fact this only works on people in the dark essentially negates it usability. Also, even if that fact weren't considered, you can always use any sort of stun you possess on the Nightmare, who is completely and utterly susceptible to them.

CMDR-Gungnir commented:
> It’d be neat if phobia’s also caused terror build up

Maybe someone could make a PR for that once the status effect exists.

MrDoomBringer commented:
I don't really see the need for this. Once you're in the dark without a light you're already screwed. Nightmare needs an aoe-destroy-nearby-lightsources action or something to counter people dragging in 15 lamps in a box to fight him, not more buffs for already being in the dark.

githubuser4141 commented:
Does this affect people with NVGs, and people in enclosed mechs? I think people in enclosed mechs should still be *spooked*, but be shielded from the gradual terror buildup that a non-enclosed person would experience. Ala, you're freaked out, but you're physically isolated when you're viewing it through your mecha's sensor slits so it doesn't get to you as bad.

Fikou committed:
Merge branch 'master' into terrorize

Fikou reviewed (changes requested):
your terrorize spell hard deletes

Rhials committed:
properly lays this ghost to rest

Fikou commented on code/modules/antagonists/nightmare/nightmare_organs.dm:
```diff
@@ -102,5 +102,30 @@
+	if(reciever.mind?.has_antag_datum(/datum/antagonist/nightmare)) //Only a TRUE NIGHTMARE is worthy of using this ability
+		terrorize_spell.Grant(reciever)
+
+/obj/item/organ/internal/eyes/night_vision/nightmare/Remove(mob/living/carbon/eye_owner, special)
+	QDEL_NULL(terrorize_spell)
+	return ..()
```
no, only delete the spell if the eyes are deleted (in Destroy), this will runtime if the eyes are removed and inserted into someone

Rhials committed:
try this

JohnFulpWillard commented on code/datums/status_effects/debuffs/terrified.dm:
```diff
@@ -0,0 +1,156 @@
+		owner.adjust_eye_blur_up_to(10 SECONDS * delta_time, 10 SECONDS)
+		if(prob(5)) //We have a little panic attack. Consider it GENTLE ENCOURAGEMENT to start running away.
+			freak_out(PANIC_ATTACK_TERROR_AMOUNT)
+			owner.visible_message(span_warning("[owner] drops to the floor for a moment, clutching their chest."), span_alert("Your heart lurches in your chest. You can't take much more of this!"), span_hear("You hear a grunt."))
```
```suggestion
			owner.visible_message(
				span_warning("[owner] drops to the floor for a moment, clutching their chest."),
				span_alert("Your heart lurches in your chest. You can't take much more of this!"),
				span_hear("You hear a grunt."),
			)
```

JohnFulpWillard commented on code/datums/status_effects/debuffs/terrified.dm:
```diff
@@ -0,0 +1,156 @@
+		owner.remove_fov_trait(id, FOV_270_DEGREES)
+
+	if(terror_buildup >= TERROR_HEART_ATTACK_THRESHOLD) //You should only be able to reach this by actively terrorizing someone
+		owner.visible_message(span_warning("[owner] clutches [owner.p_their()] chest for a moment, then collapses to the floor."), span_alert("The shadows begin to creep up from the corners of your vision, and then there is nothing..."), span_hear("You hear something heavy collide with the ground."))
```
```suggestion
		owner.visible_message(
			span_warning("[owner] clutches [owner.p_their()] chest for a moment, then collapses to the floor."), 
			span_alert("The shadows begin to creep up from the corners of your vision, and then there is nothing..."), 
			span_hear("You hear something heavy collide with the ground."),
		)
```

JohnFulpWillard commented on code/datums/status_effects/debuffs/terrified.dm:
```diff
@@ -0,0 +1,156 @@
+	if(is_species(hugger, /datum/species/shadow/nightmare)) //hey wait a minute, that's not a comforting, friendly hug!
+		if(check_surrounding_darkness())
+			addtimer(CALLBACK(src, PROC_REF(freak_out), HUG_TERROR_AMOUNT))
+			owner.visible_message(span_warning("[owner] recoils in fear as [hugger] waves [hugger.p_their()] arms and shrieks at [owner.p_them()]!"), span_boldwarning("The shadows lash out at you, and you drop to the ground in fear!"), span_hear("You hear someone shriek in fear. How embarassing!"))
```
```suggestion
			owner.visible_message(
				span_warning("[owner] recoils in fear as [hugger] waves [hugger.p_their()] arms and shrieks at [owner.p_them()]!"),
				span_boldwarning("The shadows lash out at you, and you drop to the ground in fear!"),
				span_hear("You hear someone shriek in fear. How embarassing!"),
			)
```

JohnFulpWillard commented on code/datums/status_effects/debuffs/terrified.dm:
```diff
@@ -0,0 +1,156 @@
+
+	if(is_species(hugger, /datum/species/shadow/nightmare)) //hey wait a minute, that's not a comforting, friendly hug!
+		if(check_surrounding_darkness())
+			addtimer(CALLBACK(src, PROC_REF(freak_out), HUG_TERROR_AMOUNT))
```
Do we want this to be a chain-stun? If not, then maybe it will only affect people standing up?

JohnFulpWillard replied:
That's fair, just wanted to make sure it was thought-out.

Rhials committed:
visible message stacking stuff

Rhials committed:
move it to the mind

having these on the eyes was one of the first things I did back when this was a much much different idea

Rhials committed:
outta here

Rhials committed:
work dammit

JohnFulpWillard commented on code/datums/status_effects/debuffs/terrified.dm:
```diff
@@ -0,0 +1,165 @@
+/datum/status_effect/terrified/proc/comfort_owner(datum/source, mob/living/hugger)
+	SIGNAL_HANDLER
+
+	if(is_species(hugger, /datum/species/shadow/nightmare)) //hey wait a minute, that's not a comforting, friendly hug!
```
```suggestion
	if(isnightmare(hugger)) //hey wait a minute, that's not a comforting, friendly hug!
```

JohnFulpWillard commented on code/datums/status_effects/debuffs/terrified.dm:
```diff
@@ -0,0 +1,165 @@
+		span_nicegreen("You feel yourself calm down as [hugger] gives you a reassuring hug."),
+		span_hear("You hear shuffling and a sigh of relief."),
+	)
+	return
```
Don't need to return here, it's end of the proc

Rhials committed:
helper helper helper HELP HELP HELP

Rhials committed:
Merge remote-tracking branch 'upstream/master' into terrorize
